Microsoft.Toolkit.HighPerformance/Helpers/Internals/BitOperations.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Contracts;
+using System.Runtime.CompilerServices;
+#if NETCOREAPP3_1
+using static System.Numerics.BitOperations;
+#endif
+
+namespace Microsoft.Toolkit.HighPerformance.Helpers.Internals
+{
+    /// <summary>
+    /// Utility methods for intrinsic bit-twiddling operations. The methods use hardware intrinsics
+    /// when available on the underlying platform, otherwise they use optimized software fallbacks.
+    /// </summary>
+    internal static class BitOperations
+    {
+        /// <summary>
+        /// Rounds up an <see cref="int"/> value to a power of 2.
+        /// </summary>
+        /// <param name="x">The input value to round up.</param>
+        /// <returns>The smallest power of two greater than or equal to <paramref name="x"/>.</returns>
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int RoundUpPowerOfTwo(int x)
+        {
+#if NETCOREAPP3_1
```

Comment:
This will work I assume on .NET 5 too when we get there?

Reply:
Yup, it should pick that up automatically as the `NETCOREAPP3_1` tag is inherited (see [here](https://docs.microsoft.com/en-us/dotnet/standard/frameworks)). This is also the reason for some of the reshuffling done in #3356, as right now .NET 5 will automatically just pick up the .NET Core version, so in case you're actually running .NET 5 from Mono that could cause issues (eg. in the fast array accessors, which have been modified accordingly in that PR). In this particular case though this is fine, as this just uses normal public APIs and doesn't do any tricks that specifically require to be on CoreCLR to work 👍